core/service/src/lib.rs

```diff
@@ -261,20 +229,17 @@ impl<Components: components::Components> Service<Components> {
 		let network_status_sinks = Arc::new(Mutex::new(Vec::new()));
 
 		let keystore_authority_key = AuthorityKeyProvider {
-			roles: config.roles,
-			password: config.password.clone(),
-			keystore: keystore.map(Arc::new),
+			keystore: keystore.clone(),
 		};
-
 		#[allow(deprecated)]
 		let offchain_storage = client.backend().offchain_storage();
 		let offchain_workers = match (config.offchain_worker, offchain_storage) {
 			(true, Some(db)) => {
 				Some(Arc::new(offchain::OffchainWorkers::new(
 					client.clone(),
 					db,
-					keystore_authority_key.clone(),
-					config.password.clone(),
+					keystore_authority_key,
+					"".to_string().into(),
```

Comment:
I think it would be better to introduce a CLI flag to pass the password for offchain worker keys to prevent keeping them in raw form in the DB.
I'm fine if it''s not part of this PR, but please create an issue and add a TODO in the code.

Reply:
do offchain workers require a password? In this PR I'm proposing not ever persisting a key to the filesystem as for session keys we don't want that. Currently session keys are the only ones stored in the keystore

Reply:
Offchain workers can create (and persist) their own crypto keys. That ends up in the local database, and was additionally protected by the same password that user passed to unlock authority keys.
Now that I think of it, I should have really introduced a separate CLI flag right from the start.